lib/AST/ASTImporter.cpp

```diff
@@ -7554,7 +7560,10 @@ Decl *ASTImporter::Imported(Decl *From, Decl *To) {
   if (From->isImplicit()) {
     To->setImplicit();
   }
+  To->IdentifierNamespace = From->IdentifierNamespace;
+  
```

Comment:
Copy namespace only at first time the node was imported (normally immediate after creating it).
The namespace can be adjusted later by functions like `setObjectOfFriendDecl` and this change should not overwritten by a possible next call to `Imported` or `Import`.

Reply:
:+1: 
OK, but it is really important that we should keep this rule: " A ::Create is immediately followed by an Imported() " . I already planned to bring in exactly these changes from #324.